copier/vcs.py

```diff
@@ -15,7 +15,7 @@
 from .tools import TemporaryDirectory
 from .types import OptBool, OptStr, StrOrPath
 
-GIT_PREFIX = ("git@", "git://", "git+")
+GIT_PREFIX = ("git@", "git://", "git+", "https://github.com", "https://gitlab.com")
```

Comment:
If we reduce it just to https:// then https://google.com would come through as a valid repo. I'm new to contributing and perhaps I'm just not fully understanding the scope of this project. I thought that the intent was to only return a valid git repo URL. I specifically chose those two sites based on your comment, "I guess adding the exception for github and gitlab would be acceptable." in issue #217

Reply:
Have you tried? I was assuming git will fail if we pass it `https://google.com` and that exception should be explicit enough.

If you prefer to keep the scope more targeted, at least add the trailing `/` to the sites (to avoid false positives on `https://gitlab.com.evil.example.com/`).

Reply:
I completely agree. In fact, I actually added the trailing `/` in my [b5ddf6b](https://github.com/copier-org/copier/pull/677/commits/b5ddf6bf3336697faa9edc7840be600d6cf7dab6) commit. In hindsight, I should have made two commits: one for the trailing `/` and one for the test case addition.

Concerning whether or not I've tried, I'm going to be straight with you - I can't get copier working on things I think it should work for. I suspect that it's a me problem, not a program problem. This may have led me to being more robust than is strictly necessary based on not being sure of how the downstream would handle returning a URL that is not a valid git repo.

Reply:
> I'm going to be straight with you - I can't get copier working on things I think it should work for.

Thanks for your honesty. I don't claim Copier is perfect, but it's certainly quite different from other scaffolding managers out there. We have a forum where you can ask anything you want, give it a try!